### PR TITLE
Add support for customizing row and column count in home grid

### LINKED ISSUE #60: [Feature Request] Customizable number of rows/columns
I prefer grid of size 6x9 and it is kind of a deal breaker for me to have a fixed grid like it is right now.
This typically also implies ability to change icon size as for larger grids you have to have them smaller (unless application adjusts them adequately automatically).

### PR DESCRIPTION
**Pull request by esensar**
This closes #60 

## TODO

- [x] Finalize strings and move them to strings.xml
- [x] Wait for https://github.com/SimpleMobileTools/Simple-Commons/pull/1751 and update Simple-Commons ref

### CODE REVIEW

**Comment by esensar** [comment]
This is ready for testing, only strings need to be finalized and then updated for translations.

File: app/src/main/res/layout/activity_settings.xml [hunk 1]
```diff
@@ -170,6 +170,65 @@
...
+                android:layout_width="match_parent"
+                android:layout_height="wrap_content"
+                android:text="Home screen" />
+
+            <RelativeLayout
+                android:id="@+id/settings_home_screen_row_count_holder"
+                style="@style/SettingsHolderTextViewOneLinerStyle"
+                android:layout_width="match_parent"
+                android:layout_height="wrap_content"
+                android:background="@drawable/ripple_bottom_corners">
```
Comment (tibbi) [inline app/src/main/res/layout/activity_settings.xml:190]: we dont use these corners anymore since settings redesign

**Comment by tibbi** [comment]
Replace `Home screen row count` and column count with just `Row count` and `Column count`, Home screen label makes it clear what is it about.

**Comment by tibbi** [comment]
clickin on "Row count", the radio items say columns like "2 columns" etc instead of rows, added a new string `row_counts` for it

**Comment by esensar** [comment]
Made the updates and opened up https://github.com/SimpleMobileTools/Simple-Commons/pull/1751 for adding `row_count` label.

Regarding the dropping of items on the grid, maybe best to do it in a separate PR as you said, since I think that issue was present before these changes.

**Comment by tibbi** [comment]
create a separate PR for it, also if you have any other improvements, those can be sent as a separate PR. I miss for example some small bounde effect at pulling the app drawer up :)

File: app/src/main/res/layout/activity_settings.xml [hunk 2]
```diff
@@ -171,6 +171,22 @@
 
             </RelativeLayout>
 
+            <RelativeLayout
+                android:id="@+id/settings_drawer_search_holder"
+                style="@style/SettingsHolderTextViewOneLinerStyle"
+                android:layout_width="match_parent"
+                android:layout_height="wrap_content"
+                android:background="@drawable/ripple_bottom_corners">
```
Comment (tibbi) [inline app/src/main/res/layout/activity_settings.xml]: do not use bottom corners, just full ripple. Main branch should have latest fixes already.

Comment (esensar) [reply app/src/main/res/layout/activity_settings.xml]: Forgot to update it on this branch too

**Comment by esensar** [comment]
Regarding the dropping of items on the grid, I have only been able to reproduce that when the icon is supposed to be dropped on top of another icon. I guess until we add support for folders (if we decide to do that), there is not much to be done about it.

**Comment by tibbi** [comment]
try using 2x2 grid and dropping an item inbetween, it is easy to reproduce. When you drag the icon on the home screen and there is no shadow below it, it breaks.

**Comment by esensar** [comment]
> try using 2x2 grid and dropping an item inbetween, it is easy to reproduce. When you drag the icon on the home screen and there is no shadow below it, it breaks.

Yeah, got it
It was reproducible only when just changing the dimensions. I guess when I tested I had 2x2 set before starting the app already, so it worked as expected.

File: app/src/main/kotlin/com/simplemobiletools/launcher/fragments/WidgetsFragment.kt [hunk 3]
```diff
@@ -238,6 +239,11 @@ class WidgetsFragment(context: Context, attributeSet: AttributeSet) : MyFragment
     }
 
     override fun onWidgetLongPressed(appWidget: AppWidget) {
+        if (appWidget.heightCells > context.config.homeRowCount - 1 || appWidget.widthCells > context.config.homeColumnCount) {
+            context.toast("Widget is too big for current home screen size")
```
Comment (tibbi) [inline app/src/main/kotlin/com/simplemobiletools/launcher/fragments/WidgetsFragment.kt:243]: guess we should have a string for this

Comment (esensar) [reply app/src/main/kotlin/com/simplemobiletools/launcher/fragments/WidgetsFragment.kt]: Forgot about that one. Updated

**Comment by tibbi** [comment]
should be good now, thanks

**Pull request merged**